da/celestia/celestia.go

```diff
@@ -129,9 +130,15 @@ func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context
 func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64) da.ResultRetrieveBlocks {
 	data, err := c.client.NamespacedData(ctx, c.namespaceID, dataLayerHeight)
 	if err != nil {
+		var code da.StatusCode
+		if strings.Contains(err.Error(), da.ErrDataNotFound.Error()) || strings.Contains(err.Error(), da.ErrEDSNotFound.Error()) {
```

Comment:
```suggestion
		if errors.Is(err, da.ErrDataNotFound) || errors.Is(err, da.ErrEDSNotFound) {
```

Reply:
it doesn't work as the source is not using the same Err object. for `errors.Is` to work, we need common error types extracted from celestia-node so that we can reuse the error objects.

Reply:
ah, now I see what you did :+1: